app/controllers/concerns/application_shared.rb

```diff
@@ -33,7 +33,7 @@ def set_taxonomy
           if params["#{taxonomy}_id"].blank? # the key is present and explicitly set to nil which indicates "any" context
             determined_taxonomy = nil
           else
-            determined_taxonomy = available.where(:id => params["#{taxonomy}_id"]).first
+            determined_taxonomy = scope_by_resource_id(available, params["#{taxonomy}_id"]).first
```

Comment:
This method is defined in `Api::BaseController` you can't use it in `ApplicationShared` as it is included in both UI and API controllers.
Why it did not fail locally, I've got no clue, but it should fail